src/liballoc/collections/btree/node.rs

```diff
@@ -158,29 +158,32 @@ impl<K, V> InternalNode<K, V> {
 }
 
 /// An owned pointer to a node. This basically is either `Box<LeafNode<K, V>>` or
-/// `Box<InternalNode<K, V>>`. However, it contains no information as to which of the two types
-/// of nodes is actually behind the box, and, partially due to this lack of information, has no
-/// destructor.
+/// `Box<InternalNode<K, V>>` or `Box<NodeHeader<(), ())` (i.e. EMPTY_ROOT_NODE).
+/// However, it contains no information as to which of the three types of nodes
+/// is actually behind the box, and, partially due to this lack of information,
+/// has no destructor.
 struct BoxedNode<K, V> {
-    ptr: Unique<LeafNode<K, V>>,
+    ptr: Unique<NodeHeader<K, V>>,
 }
 
 impl<K, V> BoxedNode<K, V> {
     fn from_leaf(node: Box<LeafNode<K, V>>) -> Self {
-        BoxedNode { ptr: Box::into_unique(node) }
+        unsafe {
+            BoxedNode { ptr: Unique::new_unchecked(Box::into_raw(node) as *mut NodeHeader<K, V>) }
```

Comment:
I see that this is consistent with `from_internal` but I'm tempted to suggest rewriting both into `Box::into_unique(node).cast()`  to get rid of the gratuitous `unsafe`.

Reply:
Done, and alloc tests and Miri are happy